### PR TITLE
refactor: hidden fields instead of internal fields

### LINKED ISSUE #1299: [engine] refactor the way we store internal fields to use a weak map
https://github.com/salesforce/lwc/blob/37f8cf2a3461da54d8c5a9cdfe84bc949758097a/packages/%40lwc/engine/src/framework/vm.ts#L616

### PR DESCRIPTION
**Pull request by ekashida**
## Details

Fixes #1299

## Does this PR introduce breaking changes?

* ✅ `No, it does not introduce breaking changes.`

### CODE REVIEW

**Comment by ekashida** [comment]
@caridy @ravijayaramappa There were no comments from best, does that mean there was no perf regression? Also, any ideas why the perf tests require the vm to be set internally in vm.ts?

File: packages/@lwc/engine/src/framework/base-lightning-element.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ import {
     getComponentAsString,
     getTemplateReactiveObserver,
 } from './component';
-import { setInternalField, setHiddenField } from '../shared/fields';
+import { setHiddenField, setInternalField } from '../shared/fields';
```
[the remark this answers is not in the document]
Comment (ekashida) [reply packages/@lwc/engine/src/framework/base-lightning-element.ts]: Sorry, I mentioned above that it was in `vm.ts` but it was actually in `def.ts`:
```
/**
 * EXPERIMENTAL: This function provides access to the component constructor,
 * given an HTMLElement. This API is subject to change or being removed.
 */
export function getComponentConstructor(elm: HTMLElement): ComponentConstructor | null {
    let ctor: ComponentConstructor | null = null;
    if (elm instanceof HTMLElement) {
        const vm = getInternalField(elm, ViewModelReflection);
        if (!isUndefined(vm)) {
            ctor = vm.def.ctor;
        }
    }
    return ctor;
}
```
I'm not sure why, but our perf tests blow up when we access the vm using `getHiddenField` here.

[the remark this answers is not in the document]
Comment (caridy) [reply packages/@lwc/engine/src/framework/base-lightning-element.ts]: hmm, let me look at it.

**Comment by ravijayaramappa** [comment]
@ekashida I think BEST summary is posted only if there is a significant regression. 
You can view the results by clicking on the Details link.
Screen Shot 2019-08-30 at 10 21 09 AM

**Comment by caridy** [comment]
No @ravijayaramappa it is indeed failing, I have tried in another PR:

```
Error: Benchmark parse error.
TypeError: Cannot read property 'id' of undefined
    at tmpl (http://127.0.0.1:38538/wc-append-1k.benchmark.js:9209:32)
    at evaluateTemplate (http://127.0.0.1:38538/wc-append-1k.benchmark.js:6804:27)
    at http://127.0.0.1:38538/wc-append-1k.benchmark.js:6934:20
    at ReactiveObserver.observe (http://127.0.0.1:38538/wc-append-1k.benchmark.js:5160:11)
    at vmBeingRendered (http://127.0.0.1:38538/wc-append-1k.benchmark.js:6932:16)
    at runWithBoundaryProtection (http://127.0.0.1:38538/wc-append-1k.benchmark.js:8442:9)
    at invokeComponentRenderMethod (http://127.0.0.1:38538/wc-append-1k.benchmark.js:6928:7)
    at renderComponent (http://127.0.0.1:38538/wc-append-1k.benchmark.js:7036:22)
    at rehydrate (http://127.0.0.1:38538/wc-append-1k.benchmark.js:8065:26)
    at rerenderVM (http://127.0.0.1:38538/wc-append-1k.benchmark.js:7967:7)
  ERROR     Error: Benchmark couldn't finish running. 
    at Socket.socket.on (/home/circleci/lwc/node_modules/@best/runner-hub/build/HubClient.js:84:24)
    at Socket.Emitter.emit (/home/circleci/lwc/node_modules/component-emitter/index.js:133:20)
    at Socket.onevent (/home/circleci/lwc/node_modules/socket.io-client/lib/socket.js:278:10)
    at Socket.onpacket (/home/circleci/lwc/node_modules/socket.io-client/lib/socket.js:236:12)
    at Manager.<anonymous> (/home/circleci/lwc/node_modules/component-bind/index.js:21:15)
    at Manager.Emitter.emit (/home/circleci/lwc/node_modules/component-emitter/index.js:133:20)
    at Manager.ondecoded (/home/circleci/lwc/node_modules/socket.io-client/lib/manager.js:345:8)
    at Decoder.<anonymous> (/home/circleci/lwc/node_modules/component-bind/index.js:21:15)
    at Decoder.Emitter.emit (/home/circleci/lwc/node_modules/component-emitter/index.js:133:20)
    at Decoder.add (/home/circleci/lwc/node_modules/socket.io-client/node_modules/socket.io-parser/index.js:251:12)
error Command failed with exit code 1.```

I will investigate.

**Comment by ekashida** [comment]
#1485

**Pull request closed without merging**